### PR TITLE
Remove unreachable code

### LINKED ISSUE #51838: Assembly.ReflectionOnlyLoad(...) is obsoleted, consider remove the unused reference in mono
As part of https://github.com/dotnet/runtime/issues/50529 we have obsoleted several APIs not supported in .Net Core which caused a build warning in mono CoreClr 

I see the API calling `Assembly.ReflectionOnlyLoad(...)` is always called with a value of `false` for `bool reflectionOnly` parameter, which means that reference will never actually used. We might want to remove this parameter and related code section from https://github.com/dotnet/runtime/blob/90f97ac50154ee23449f273c3a8d13dfe985815b/src/mono/System.Private.CoreLib/src/System/RuntimeTypeHandle.cs#L290-L291

_Originally posted by @buyaa-n in https://github.com/dotnet/runtime/pull/50941#discussion_r619417207_

### PR DESCRIPTION
**Pull request by fanyang-mono**
Fixed #51838

### CODE REVIEW

**Comment by SamMonoRT** [comment]
@fanyang-mono - are there any associated tests that can be removed/disabled ?

File: src/mono/System.Private.CoreLib/src/System/RuntimeTypeHandle.cs [hunk 1]
```diff
@@ -287,7 +287,7 @@ internal static bool IsTypeDefinition(RuntimeType type)
         private static extern RuntimeType internal_from_name(string name, ref StackCrawlMark stackMark, Assembly? callerAssembly, bool throwOnError, bool ignoreCase, bool reflectionOnly);
```
Comment (CoffeeFlux) [inline src/mono/System.Private.CoreLib/src/System/RuntimeTypeHandle.cs:287]: I think this PR is incomplete as-is, and we should fix this icall up as well. Reflection-only loads should all be excised from the native side, so these bits in managed are the only remaining work and I'd like to see it all done together.

**Comment by CoffeeFlux** [comment]
> are there any associated tests that can be removed/disabled ?

I doubt it - reflection-only loads have never been a part of netcore, so this has always been dead code.

File: src/mono/System.Private.CoreLib/src/System/RuntimeTypeHandle.cs (hunk 1 above)
Comment (fanyang-mono) [reply src/mono/System.Private.CoreLib/src/System/RuntimeTypeHandle.cs]: Fixed! Thanks for the suggestion.

**Review by vargaz — APPROVED** [review]

**Review by CoffeeFlux — APPROVED** [review]

**Pull request merged**